### DISCUSSION TITLE
Color mode transition

### DISCUSSION THREAD

**Discussion by rikusen0335** [chakra-ui/chakra-ui#6095] [Q&A] [answered]
As title says, I couldn't finally find a solution.

I found [this](https://github.com/chakra-ui/chakra-ui/discussions/4501) and tried:
```ts
import { ChakraProvider, extendTheme } from '@chakra-ui/react'

const theme = extendTheme({
    body: {
        transitionProperty: 'all',
        transitionDuration: 'normal',
    },
})

function App({ Component, pageProps: { session, ...pageProps } }: AppProps) {
    return (
         <ChakraProvider theme={theme}>
             <Component {...pageProps} />
         </ChakraProvider>
    )
}
```

but it seems not working as I expected.

Please give me an idea about this. Thanks.

**Comment by hugobqd** [comment, marked as answer]
Hi!
There is no 'body' key in the theme object, it should be written in theme.styles.global like that:

```js
const theme = extendTheme({
  styles: {
    global: {
      body: {
        transitionProperty: "all",
        transitionDuration: "normal",
      },
    },
  },
});
```

But note that there are already `transition-property: background-color` & `transition-duration: var(--chakra-transition-duration-normal)` by default on the body.

**Reply by rikusen0335** [reply]
Thank you for the information.

It seems overwriting the variables are fine, but still it is not applied transition as you said:
> But note that there are already transition-property: background-color & transition-duration: var(--chakra-transition-duration-normal) by default on the body.

What could be a cause of this?

**Reply by hugobqd** [reply]
can you reproduce your issue in a sandbox ?

**Reply by rikusen0335** [reply]
Probably. Wait for it

**Reply by rikusen0335** [reply]
https://codesandbox.io/s/chakra-ui-colormode-lr2ewd?file=/src/index.tsx
Here it is

**Reply by hugobqd** [reply]
Founded 😅
There is a dedicated config that prevent transition by default on toggle mode :

https://chakra-ui.com/guides/migration#color-more-transition
> Ensure the transition between light/dark modes happens instantly without transition. This helps to avoid a weird UX when switch modes for elements with different transition definition on the page.
To opt out of this feature, set theme.config.disableTransitionOnChange to false.

So to allow transition on toggle mode you have to specify it :

```js
const theme = extendTheme({
  styles: {
    global: {
      body: {
        transitionProperty: "all",
        transitionDuration: "normal"
      }
    }
  },
  config: {
    disableTransitionOnChange: false
  }
});
```

[Modified Sandbox](https://codesandbox.io/s/chakra-ui-colormode-forked-pgzxli?file=/src/index.tsx)

**Reply by rikusen0335** [reply]
Oh, this is exactly what I was looking for.
It looks working now, thank you so much.

**Reply by 0x33dm** [reply]
@hugobqd this is awesome! i wonder where i can see all those tokens? for instance "ultra-slow" or "normal"? where is this set?

I have opened a thread asking for a list with all of them: https://github.com/chakra-ui/chakra-ui/discussions/8413

**Reply by rikusen0335** [reply]
@0x33dm 
Have you looked for d.ts or typing files? I think tokens should be written in there

**Reply by 0x33dm** [reply]
yeah tokens were not there, i have only next-env.d.ts as i started from the "chakra-ui" example on next.js repo.

I just noticed now tough there is an option to use chakra cli to generate those?

https://chakra-ui.com/docs/styled-system/cli

did you have to generate them yourself, is that right?

**Reply by 0x33dm** [reply]
@rikusen0335 i created an issue explaining [my problem here](https://github.com/chakra-ui/chakra-ui/issues/8430)
